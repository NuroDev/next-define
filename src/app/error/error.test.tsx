import { describe, it, expect, expectTypeOf } from "vitest";
import renderer from "react-test-renderer";

import { defineError } from ".";

import type { FC } from "react";

import type { NextErrorPageProps } from "./error.action";

describe("error", () => {
  it("Create a basic error page component", () => {
    const Component = defineError(({ error, reset }) => {
      expect(error).toBeDefined();
      expect(error).toBeInstanceOf(Error);
      expect(reset).toBeDefined();
      expect(reset).toBeInstanceOf(Function);
      expectTypeOf(reset).toMatchTypeOf<() => void>();

      return (
        <>
          <h1>Error Page</h1>
          <p>{error.message}</p>
          <button onClick={reset} type="button">
            Retry
          </button>
        </>
      );
    });

    const ComponentJson = renderer
      .create(<Component error={new Error()} reset={() => {}} />)
      .toJSON();
    expect(ComponentJson).toBeDefined();
    expect(ComponentJson).toMatchSnapshot();
    expectTypeOf(Component).toMatchTypeOf<FC<NextErrorPageProps>>();
  });
});
