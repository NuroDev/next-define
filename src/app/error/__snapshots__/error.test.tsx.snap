// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`error > Create a basic error page component 1`] = `
[
  <h1>
    Error Page
  </h1>,
  <p />,
  <button
    onClick={[Function]}
    type="button"
  >
    Retry
  </button>,
]
`;
